tools/azsdk-cli/Azure.Sdk.Tools.Cli/Prompts/Templates/LogAnalysisTemplate.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.Sdk.Tools.Cli.Prompts.Templates;
+
+/// <summary>
+/// Template for log analysis prompts, particularly for Azure Pipeline failures.
+/// This template provides structured analysis of logs with JSON output requirements.
+/// </summary>
+public class LogAnalysisTemplate : BasePromptTemplate
+{
+    public override string TemplateId => "log-analysis";
+    public override string Version => "1.0.0";
+    public override string Description => "Analyze logs for failures, errors, and provide structured diagnostics";
+
+
+    protected override IEnumerable<string> RequiredParameters => new[] { "log_content" };
+    protected override IEnumerable<string> OptionalParameters => new[] 
+    { 
+        "log_type", 
+        "focus_areas",
+        "error_patterns",
+        "output_format"
+    };
+
+    protected override string BuildTaskInstructions(IPromptContext context)
```

Comment:
I like that we have these all split into "instructions, constraints,requirements,examples". Do we need to have separate classes/interfaces for everything. Could the structure of the prompt/baseprompt/template etc. be flattened down?